pkg/xray/provider_test.go

```diff
@@ -42,16 +44,82 @@ func getTestResty(t *testing.T) *resty.Client {
 }
 
 func testAccPreCheck(t *testing.T) {
-	restyClient := getTestResty(t)
-	// TODO check the payload and make sure it's the right license type
-	_, err := restyClient.R().Get("/artifactory/api/system/licenses/")
-	if err != nil {
-		t.Fatal(err)
-	}
 	ctx := context.Background()
-	provider, _ := testAccProviders["xray"]()
+	provider, _ := testAccProviders()["xray"]()
 	oldErr := provider.Configure(ctx, terraform.NewResourceConfigRaw(nil))
 	if oldErr != nil {
 		t.Fatal(oldErr)
 	}
 }
+
+// Create a local repository with Xray indexing enabled. It will be used in the tests
+func testAccCreateRepos(t *testing.T, repos []string) {
+	restyClient := getTestResty(t)
+
+	type Repository struct {
+		Rclass    string `json:"rclass"`
+		XrayIndex bool   `json:"xrayIndex"`
+	}
+
+	repository := Repository{}
+	repository.Rclass = "local"
+	repository.XrayIndex = true
+	for _, repo := range repos {
+		response, errRepo := restyClient.R().SetBody(repository).Put("artifactory/api/repositories/" + repo)
+		repoExists := strings.Contains(fmt.Sprint(errRepo), "Case insensitive repository key already exists")
+		if !repoExists && response.StatusCode() != http.StatusOK {
+			t.Fatal(errRepo)
+		}
+	}
+}
+
+func testAccDeleteRepos(t *testing.T, repos []string) (*resty.Response, error) {
+	restyClient := getTestResty(t)
+	for _, repo := range repos {
+		response, errRepo := restyClient.R().Delete("artifactory/api/repositories/" + repo)
+		if response.StatusCode() != http.StatusOK {
+			t.Fatal(errRepo)
+		}
+	}
+	return nil, nil
+}
+
+// Create a set of builds or a single build, add the build into the Xray indexing configuration, to be able to add it to
+// the xray watch
+func testAccCreateBuilds(t *testing.T, builds []string) {
+	restyClient := getTestResty(t)
+
+	type BuildBody struct {
+		Version string `json:"version"`
+		Name    string `json:"name"`
+		Number  string `json:"number"`
+		Started string `json:"started"`
+	}
+
+	type XrayIndexBody struct {
+		Names []string `json:"names"`
+	}
+
+	for _, build := range builds {
+		buildBody := BuildBody{
+			Version: "1.0.1",
+			Name:    build,
+			Number:  "28",
+			Started: "2021-10-30T12:00:19.893+0300",
+		}
+		respCreateBuild, errCreateBuild := restyClient.R().SetBody(buildBody).Put("artifactory/api/build")
+		if respCreateBuild.StatusCode() != http.StatusNoContent {
+			t.Fatal(errCreateBuild)
+		}
+	}
+
+	xrayIndexBody := XrayIndexBody{
+		Names: builds,
+	}
+
+	respAddIndexBody, errAddIndexBody := restyClient.R().SetBody(xrayIndexBody).Post("xray/api/v1/binMgr/builds")
+	if respAddIndexBody.StatusCode() != http.StatusOK {
+		t.Fatal(errAddIndexBody)
```

Comment:
I am not sure about all this `fatal` - I mean, there is a lot of code that let's you return `error` just fine

Reply:
I see it [here](https://github.com/hashicorp/terraform-provider-hashicups/blob/1adc56591a26bf384f08c35ff5e3d88aea30b206/hashicups/provider_test.go#L32)
and [here](https://github.com/jfrog/terraform-provider-artifactory/blob/a0d2e8eddfd79e68f6304845e13efe556b29bc5a/pkg/artifactory/provider_test.go#L25)
Should we replace it everywhere?

Reply:
I've replaced `Fatal` with `Error` in all tests